frontend/src/metabase/public/LocaleProvider.tsx

```diff
@@ -15,7 +15,10 @@ export const LocaleProvider = ({
     if (locale) {
       setIsLoadingLocale(true);
       setLocaleHeader(locale);
-      loadLocalization(locale).then(() => setIsLoadingLocale(false));
+      loadLocalization(locale).then(translatedObject => {
+        setIsLoadingLocale(false);
+        setUserLocale(translatedObject);
```

Comment:
See this [original comment](https://github.com/metabase/metabase/pull/49441#discussion_r1827967395) on why I did this.